cmd/zoekt-dynamic-indexserver/main.go

```diff
@@ -222,23 +221,23 @@ func emptyDirectory(dir string) error {
 }
 
 func parseOptions() Options {
-	dataDir := flag.String("data_dir", "", "directory holding all data.")
-	indexDir := flag.String("index_dir", "", "directory holding index shards. Defaults to $data_dir/index/")
-	timeout := flag.Duration("index_timeout", time.Hour, "kill index job after this much time")
+	repoDir := flag.String("repo_dir", "", "directory holding cloned repos.")
+	indexDir := flag.String("index_dir", "", "directory holding index shards.")
+	timeout := flag.Duration("index_timeout", time.Hour, "kill index job after this much time.")
 	listen := flag.String("listen", ":6060", "listen on this address.")
 	flag.Parse()
 
-	if *dataDir == "" {
-		log.Fatal("must set -data_dir")
+	if *repoDir == "" {
+		log.Fatal("must set -repo_dir")
 	}
 
 	if *indexDir == "" {
-		*indexDir = filepath.Join(*dataDir, "index")
+		log.Fatal("must set -index_dir")
+		*indexDir = filepath.Join(*repoDir, "index")
 	}
 
 	return Options{
-		dataDir:      *dataDir,
-		repoDir:      filepath.Join(*dataDir, "repos"),
```

Comment:
I got rid of this extra `repoDir` that was here before. This previous `dataDir` was only used for storing repos anyway. The main reason this executable is separating repos from index is that repos will be very large and mostly used like a cache for bare repos. We may want to tweak storage on this differently to the index. The separation was also copied from how the other `indexserver` worked but in that one there are logs being stored somewhere else. If we introduce log files to this executable I think we'll make it again a separate argument as log storage, repo storage and index storage all have quite different needs so having the flexibility to mount them in whatever way we want is preferable.